Modules/Core/Common/include/itkImageBase.hxx

```diff
@@ -398,6 +398,38 @@ ImageBase<VImageDimension>::VerifyRequestedRegion()
 }
 
 
+template <unsigned int VImageDimension>
+bool
+ImageBase<VImageDimension>::IsCongruentImageGeometry(const ImageBase * otherImage,
+                                                     double            coordinateTolerance,
+                                                     double            directionTolerance) const
+{
+  // check that the image occupy the same physical space, and that
+  // each index is at the same physical location
+
+  // tolerance for origin and spacing depends on the size of pixel
+  // tolerance for directions a fraction of the unit cube.
+  const SpacePrecisionType coordinateTol =
+    itk::Math::abs(coordinateTolerance * this->GetSpacing()[0]); // use first dimension spacing
```

Comment:
Is there a specific reason to only use the first spacing dimension, instead of all of them? Otherwise I would rather use the smallest of them (something like `*std::min_element(m_Spacing.begin(), m_Spacing.end())`)

Reply:
Simplicity inherited from `ImageToImageFilter`'s code. Good for backwards compatibility.

Reply:
If backwards compatibility is the only reason, maybe `ITK_FUTURE_REMOVE_LEGACY` can be of help  😃